evm_arithmetization/src/generation/state.rs

```diff
@@ -326,6 +333,9 @@ pub struct GenerationState<F: Field> {
     pub(crate) memory: MemoryState,
     pub(crate) traces: Traces<F>,
 
+    /// Stale contexts you can prune from memory.
+    pub(crate) pruned_contexts: Vec<usize>,
```

Comment:
A bit of a nitpick but perhaps `stale_contexts` is a better name? After all, the key idea is to prune *memory*, not so much the contexts?

Reply:
Hmm, good point, but I started using `pruned_contexts` in several other places. I'll take inventory first.